lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -651,8 +651,14 @@ void SpecialFunctionHandler::handleNewlocale(ExecutionState &state,
   locale_t base = (locale_t)dyn_cast<ConstantExpr>(arguments[2])
                       ->getZExtValue(sizeof(locale_t) * 8);
   locale_t x = newlocale(mask, readStringAtAddress(state, arguments[1]).c_str(), base);
+#ifdef __APPLE__
+  // Cannot get sizeof _xlocale, as it is only forward declared. Approximating
   MemoryObject *p =
-      executor.addExternalObject(state, x, sizeof(__locale_struct), true);
```

Comment:
I think it might be better to just throw an error here like `terminateStateOnError("freelocale not supported on OSX")`.